packages/catlog/src/dbl/model_morphism.rs

```diff
@@ -363,4 +492,80 @@ mod tests {
         assert!(im.has_mor(&Path::single(ustr("positive"))));
         assert!(im.has_mor(&Path::single(ustr("negative"))));
     }
+
+    #[test]
+    fn validate_model_morphism() {
+        let theory = Arc::new(th_signed_category());
+        let negloop = negative_loop(theory.clone());
+        let posfeed = positive_feedback(theory.clone());
+
+        // A good map from h to itself
+        let omap: HashMap<_, _> = [(ustr("x"), ustr("x"))].into_iter().collect();
+        let mmap: HashMap<_, _> = [(ustr(""), Path::Id(ustr("negative")))].into_iter().collect();
+        let f = DiscreteDblModelMapping {
+            ob_map: omap.into(),
+            mor_map: mmap.into(),
+        };
+        let dmm = DblModelMorphism(&f, &negloop, &negloop);
+        assert!(dmm.validate().is_err());
+
+        // A bad map from h to itself that is wrong for the ob (it is in the map
+        // but sent to something that doesn't exist) and for the hom generator
+        // (not in the map)
+        let bad_ob_assign: HashMap<ustr::Ustr, Ustr> =
```

Comment:
You should be able to save a few characters here and below by writing `HashMap<_, _>`.